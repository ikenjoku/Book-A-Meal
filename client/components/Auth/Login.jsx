import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect, Link } from "react-router-dom";
import { loginAUser } from "../../actions/authActions/login";

export class Login extends Component {
  state = {
    data: {
      email: '',
      password: ''
    },
    error: ''
  }

  onFormInput = (event) => {
    event.preventDefault();
    const { data } = this.state;
    data[event.target.name] = event.target.value.trim();
    this.setState(() => ({ data }));
  }

  onSubmit = (event) => {
    event.preventDefault();
    const { data } = this.state;
    this.props.loginAUser(data)
  }

  render() {
    return (
      this.props.isLoggedIn === true ? <Redirect to='/menu' /> :
        <main className='mainContent login-bg'>
          <form className='form-wrapper' onSubmit={this.onSubmit}>
            <h2 className="center">Login</h2>
            <div>
              <input
                className="form-control"
                type="text"
                name="email"
                placeholder="Enter your email"
                value={this.state.data.email}
                onChange={this.onFormInput}
              />
            </div>
            <div>
              <input
                className="form-control"
                type="password"
                name="password"
                placeholder="Enter you password"
                value={this.state.data.password}
                onChange={this.onFormInput}
              />
            </div>
            <div>
              <input
                className="btn-control loginsubmitBtn"
                type="submit"
                value="Submit"
              />
            </div>
            <hr />
            <div className="right formText">
              <p>No Account? <Link to="/signup">Sign Up</Link> </p>
            </div>
          </form>
        </main>
    );
  }
}

Login.propTypes = {
  loginAUser: PropTypes.func.isRequired,
  isLoggedIn: PropTypes.bool.isRequired,
  authLoading: PropTypes.bool.isRequired,
}

export const mapStateToProps = state => ({
    isLoggedIn: state.authReducer.isLoggedIn,
    authLoading: state.authReducer.authLoading,
});

export default connect(mapStateToProps, { loginAUser })(Login);