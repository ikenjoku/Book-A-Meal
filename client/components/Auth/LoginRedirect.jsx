import React from 'react';
import { Redirect } from 'react-router-dom';
import notify from '../../actions/notify';

const LoginRedirect = () => {
  notify.error('Please log in to proceed');
  return (<Redirect to="/login" />);
};

export default LoginRedirect;
