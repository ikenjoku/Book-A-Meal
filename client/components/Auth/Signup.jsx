import React, { Component } from 'react';
import { signupAUser } from "../../actions/authActions/signup";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

class Signup extends Component {
    state = {
        signUpData: {
            firstname: '',
            lastname: '',
            username: '',
            email: '',
            password: '',
            confirmPassword: ''
        },
        error: ''
    }
    onInputChange = (event) => {
      const { signUpData } = this.state;
      signUpData[event.target.name] = event.target.value.trim();
      this.setState(() => ({ signUpData }));
    }

    onSubmit = (event) => {
      event.preventDefault();
      const { signUpData } = this.state;
      if(!signUpData.firstname || !signUpData.lastname || !signUpData.username || !signUpData.email 
        || !signUpData.password || !signUpData.confirmPassword){
            this.setState(() => ({ error: 'Please fill in all the fields' }));
      } else {
      this.props.signupAUser(signUpData);
      }
    }
    render() {
        return (
            this.props.isLoggedIn === true ?  <Redirect to='/menu'/> :
            <main className="signup-bg">
                {this.state.error &&  <p>{this.state.error}</p> }
                <form className="form-wrapper" onSubmit={this.onSubmit}>
                    <h2 className="center">Sign up</h2>
                    <div>
                        <input
                            className="form-control"
                            type="text"
                            name="firstname"
                            placeholder="First Name"
                            value={this.state.signUpData.firstname}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="text"
                            name="lastname"
                            placeholder="Last Name"
                            value={this.state.signUpData.lastname}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="text"
                            name="username"
                            placeholder="Preferred username"
                            value={this.state.signUpData.username}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="email"
                            name="email"
                            placeholder="Email"
                            value={this.state.signUpData.email}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="password"
                            name="password"
                            placeholder="Password"
                            value={this.state.signUpData.password}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            className="form-control"
                            type="password"
                            name="confirmPassword"
                            placeholder="Confirm password"
                            value={this.state.signUpData.confirmPassword}
                            onChange={this.onInputChange}
                        />
                    </div>
                    <div>
                        <input
                            type="submit"
                            className="btn-control loginsubmitBtn"
                            value="Submit"
                        />
                    </div>
                    <div className="right formText">
                        <p>Already have an account? <Link to="/login">Log In</Link> </p>
                    </div>
                </form>
            </main>
        );
    }
}

const mapDispatchToProps = dispatch => 
  bindActionCreators({signupAUser}, dispatch);

export default connect(null, mapDispatchToProps)(Signup);
