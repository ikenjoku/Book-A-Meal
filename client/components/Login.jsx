import React, { Component } from 'react';

import NavBar from './NavBar.jsx';

export default class Login extends Component{
  constructor(props){
    super(props)
  }

  render(){
    return  (
      <div>
      <NavBar />
      <main className='mainContent'>
          <form className='form-wrapper'>
          <h2 className="center">Login</h2>
          <div>
              <input 
                className="form-control" 
                type="text" 
                name="email" 
                placeholder="Email"
              />
            </div>
            <div>
              <input
                className="form-control"
                type="password"
                name="password"
                placeholder="Password"
              />
            </div>
            <div>
              <input
                className="btn-control loginsubmitBtn"
                type="submit"  
                value="Submit"
              />
            </div>

            <hr />
            <div className="center">
              <p>OR</p>
              <a href="" className="formText btn-google">Sign in with Google</a>
            </div>
            <div className="right formText">
              <p>No Account? <a href="signup.html">Sign Up</a> </p>
              <a href="#">Forgot your password?</a>
            </div>
          </form>
        </main>
        <footer className="footer">
          <p>Book-A-Meal &copy;2018</p>
        </footer>
      </div>
    );
  }
}