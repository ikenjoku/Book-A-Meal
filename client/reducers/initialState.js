export default {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [],
    error: null,
    isLoading: true,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders:false,
  },
};
