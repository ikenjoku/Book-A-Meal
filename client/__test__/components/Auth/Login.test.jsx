import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapStateToProps } from '../../../components/Auth/Login';
import initialState from '../../../reducers/initialState';

describe('Login Component', () => {
  let wrapper;
  const loginAUser = jest.fn();
  const isLoggedIn = false;

  test('it should render Login form correctly', () => {
    wrapper = shallow(<Login
      loginAUser={loginAUser}
      isLoggedIn={isLoggedIn}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onFormInput when input fields change value', () => {
    wrapper = shallow(<Login
      loginAUser={loginAUser}
      isLoggedIn={isLoggedIn}
    />);

    const onFormInputSpy = jest.spyOn(wrapper.instance(), 'onFormInput');
    const event = {
      preventDefault: jest.fn(),
      target: {
        value: 'chubby@gmail.com',
        name: 'email',
      },
    };
    wrapper.instance().onFormInput(event);
    expect(onFormInputSpy).toHaveBeenCalledTimes(1);
  });

  it('should handle onSubmit call when form is submitted', () => {
    wrapper = shallow(<Login
      loginAUser={loginAUser}
      isLoggedIn={isLoggedIn}
    />);

    const onSubmitSpy = jest.spyOn(wrapper.instance(), 'onSubmit');
    const event = {
      preventDefault: jest.fn(),
      target: {
        value: 'chubby@gmail.com',
        name: 'email',
      },
    };
    wrapper.instance().onSubmit(event);
    expect(onSubmitSpy).toHaveBeenCalledTimes(1);
  });

  it('should redirect to menu page if user signs in', () => {
    wrapper = shallow(<Login
      loginAUser={loginAUser}
      isLoggedIn={true}
    />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('Redirect').length).toBe(1);
    expect(wrapper.find('div').length).toBe(0);
  });

  test('should map state to props', () => {
    const ownProps = { isLoggedIn: false };
    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  });
});
