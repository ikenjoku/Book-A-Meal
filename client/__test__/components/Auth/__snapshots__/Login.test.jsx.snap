// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component it should render Login form correctly 1`] = `
<main
  className="mainContent login-bg"
>
  <form
    className="form-wrapper"
    onSubmit={[Function]}
  >
    <h2
      className="center"
    >
      Login
    </h2>
    <div>
      <input
        className="form-control"
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        placeholder="Enter you password"
        type="password"
        value=""
      />
    </div>
    <div>
      <input
        className="btn-control loginsubmitBtn"
        type="submit"
        value="Submit"
      />
    </div>
    <hr />
    <div
      className="right formText"
    >
      <p>
        No Account? 
        <Link
          replace={false}
          to="/signup"
        >
          Sign Up
        </Link>
         
      </p>
    </div>
  </form>
</main>
`;

exports[`Login Component should map state to props 1`] = `
Object {
  "authLoading": true,
  "isLoggedIn": false,
}
`;
